vizro-core/tests/unit/vizro/models/_components/form/test_range_slider.py

```diff
@@ -0,0 +1,331 @@
+        "title",
+        [
+            "test",
+            1,
+            1.0,
+            """## Test header""",
+            "",
+        ],
+    )
+    def test_valid_title(self, title):
+        slider = vm.RangeSlider(title=title)
+
+        assert slider.title == str(title)
+
+    def test_set_action_via_validator(self, test_action_function):
+        range_slider = vm.RangeSlider(actions=[vm.Action(function=test_action_function)])
+        actions_chain = range_slider.actions[0]
+
+        assert actions_chain.trigger.component_property == "value"
+
+
+class TestRangeSliderBuild:
+    """Tests model build method."""
+
+    def test_range_slider_build_default(self, expected_range_slider_default):
+        range_slider = vm.RangeSlider(id="range_slider")
+        component = range_slider.build()
+
+        result = json.loads(json.dumps(component, cls=plotly.utils.PlotlyJSONEncoder))
+        expected = json.loads(json.dumps(expected_range_slider_default, cls=plotly.utils.PlotlyJSONEncoder))
+
+        assert result == expected
+
+    def test_range_slider_build_with_optional(self, expected_range_slider_with_optional):
+        range_slider = vm.RangeSlider(min=0, max=10, step=1, value=[0, 10], id="range_slider_with_all", title="Title")
+        component = range_slider.build()
+
+        result = json.loads(json.dumps(component, cls=plotly.utils.PlotlyJSONEncoder))
+        expected = json.loads(json.dumps(expected_range_slider_with_optional, cls=plotly.utils.PlotlyJSONEncoder))
+
+        assert result == expected
+
+
+class TestCallbackMethod:
+    """Tests model callback method."""
+
+    @pytest.mark.parametrize(
+        "trigger_id, min, max, input_store, value, expected_value",
+        [
+            ("_start_value", 0, 10, [1, 9], [2, 3], (0, 10, [0, 10], (0, 10))),
+            ("_input_store", 0, 10, [1, 9], [2, 3], (1, 9, [1, 9], (1, 9))),
+            ("_end_value", 2, 8, [1, 9], [2, 3], (2, 8, [2, 8], (2, 8))),
+            ("", 2, 8, [1, 9], [2, 3], (2, 8, [2, 8], (2, 8))),
+            ("_start_value", 0, 0, [1, 9], [2, 3], (0, 0, [0, 0], (0, 0))),
+            ("_start_value", 0, 10, [1, 9], [-2, 12], (0, 10, [0, 10], (0, 10))),
+        ],
+    )
+    def test_update_slider_value_valid(self, trigger_id, min, max, input_store, value, expected_value):
+        range_slider = vm.RangeSlider(min=min, max=max)
+
+        result = range_slider._update_slider_values(
+            trigger_id=f"{range_slider.id}{trigger_id}",
+            start=min,
+            end=max,
+            input_store=input_store,
+            value=value,
+            slider=[min, max],
+        )
+        assert result == expected_value
+
+    @pytest.mark.parametrize(
+        "trigger, start_value, end_value, slider_value, input_store_value",
+        [
+            ("_text_value", None, 9, [1, 9], [1, 9]),
+        ],
+    )
+    def test_update_slider_invalid(self, trigger, start_value, end_value, slider_value, input_store_value):
```

Comment:
Could these ever happen? Like in principle, would a user be able to trigger this? If not then we don't really need to test for it. It is more important to think of every possible situation that could happen, and check if it works